Necro Government and Undead Flavor

zumbak04 opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Added Necro Government:
  - No vassal contracts, taxes and levies based on opinion.
  - Only rulers of the same religious family can inherit.
  - Dark Frenzy instead of Crown Authority.
    - By default, you're 50% of normal government but can be up to 150% by keeping high Frenzy.
    - High Frenzy gives you troop, tax, and development bonuses.
    - Frenzy decays over time so to be efficient they should do their mechanics.
  - Get Frenzy on converting provinces to their culture and religion.
  - Get Frenzy on winning armies, sieging, raiding.
  - Killed knights become yours as undead. High Frenzy increases the chance.
    - Becoming an undead, the character cuts all his previous relationships, abdicate, and becomes disinherited.
    - The character's previous relatives, friends, lovers get stress as the character died if their faith shuns or considers the undead criminals.
    - Reinforce armies much faster.
- Disinheriting undead is free if your faith shuns or considers them criminals.
- The Faiths of the Death religious family use black, purple, dark blue, white, grey, light red colors for CoAs.
- The Grim Religions can't do hunts and feasts.
- Northrend frozen landscape appears on event backgrounds in Icecrown, Storm Peaks, and Dragonblight.
- Rulers of the Scourge are called Lich Kings/Queens.

<!--
A changelog where you can describe more complicated things to other developers and testers.
-->
## Developer changelog:
- Added many triggers and effects to handle monarchic governments that should work kinda the same.
- Uncommented `debug_start_great_holy_war_interaction`.

# Tasks:

<details>
<summary>Click to expand</summary>

- [x] Create game concepts for new governments and Dark Governments
- [x] AI should disinherit undead, demons, etc. for free and often
- [x] Relocalize this:
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/122537363-88fc9c80-d036-11eb-95ee-713ef0baa81b.png)

</details>

- [x] Government resets on losing requirements
- [x] Adapt `StatecraftGetVassalType`.
- [x] Fix not having Crown Authority
- [x] Add progress to Dark Frenzy
- [x] Add Dark Frenzy icons
- [x] Add Dark Frenzy modifiers
- [x] Create game concept for Dark Frenzy
- [x] Send interface message on getting new Dark Frenzy law
- [x] Switching from Holy Light to Helya sets Necro government
- [x] Switching to Light clears Dark Frenzy
- [x] Change this background
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/122667213-b6765100-d1c2-11eb-9e60-be9c5034c7f6.png)

</details>

- [x] Dark Frenzy progress bar should call tooltip
- [x] Localize Dark Frenzy scripted effects
- [x] Explain why you can't change laws of Dark Frenzy yourself
- [x] Check authority change effect
- [x] Angerboda starting Undead
- [x] Grim Religions shouldn't have feasts, hunts
- [x] Handle death in `combat_event.1001`
- [x] Handle notification in `combat_event.1001`
- [x] Add loyalty opinion
- [x] Give stress/event if your friend/relative turned undead, #511
- [x] Check errors in battle, knight, combat, death events
- [x] Increase chance with Frenzy
- [x] Undead resets all character's relationships
- [x] Check if relatives get notification for game start undead
- [x] Check soulmate, secret lover, best friend
- [x] Event backgrounds
- [x] Replace `being_is_illegal_or_shunned_trigger` with `trait_is_shunned_or_criminal_in_faith_trigger`.
- [x] It causes longer loading?
- [x] Crusader realms (created in crusade)
- [x] Scourge shouldn't be claimed
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/125998125-9d729800-4400-4e74-abdf-1aebc42a417a.png)

</details>

- [ ] Mention that Dark Religions can't hunt and feast
- [x] Convert to Holy Light, kill, inherit by the undead, check if you get yearly Dark Frenzy

</details>

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
Start as a kvaldir in any date or as a vassal of the Scourge in the last 2 dates. Test the mentioned in the changelog features.

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/463/files?file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here are the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).

arithon commented on localization/replace/english/wc_replacements_l_english.yml:
```diff
@@ -73,4 +73,19 @@
+ 
+ UNLOCKS_CLAIM_THRONE_INTERACTION:0 "Unlocks the [claim_throne|E] Interaction #F (but there are exceptions)#!"
+ I_CAN_USE_THE_CLAIM_THRONE_INTERACTION:1 "You can use [claim_throne|E] against your [liege|E] #F (but there are exceptions)#!"
+ THEY_CAN_USE_THE_CLAIM_THRONE_INTERACTION:0 "[CHARACTER.GetShortUIName|U] can use [claim_throne|E] against [CHARACTER.GetHerHis] liege [liege|E] #F (but there are exceptions)#!"
```
```suggestion
 THEY_CAN_USE_THE_CLAIM_THRONE_INTERACTION:0 "[CHARACTER.GetShortUIName|U] can use [claim_throne|E] against [CHARACTER.GetHerHis] [liege|E] #F (but there are exceptions)#!"
```

arithon reviewed (changes requested):
Left some change requests, rest looks good

MemeWizard commented on localization/english/wc_laws_l_english.yml:
```diff
@@ -0,0 +1,27 @@
+﻿l_english:
+ dark_authority:0 "[dark_authority|E]"
+ dark_authority_0:0 "Disparate Cults"
```
![image](https://user-images.githubusercontent.com/46576860/126080055-26273da7-c5bf-49e6-bebb-80ad50f70550.png)

Are you sure it's the right word? Maybe you wanted to use "disperse" instead?

```suggestion
 dark_authority_0:0 "Disperse Cults"
```

But I would use this one instead
```suggestion
 dark_authority_0:0 "Scattered Cults"
```

arithon replied:
I like the Scattered Cults best of these alternatives :)

zumbak04 replied:
> Are you sure it's the right word?

Yes.

zumbak04 replied:
![image](https://user-images.githubusercontent.com/46384992/126084058-74153d45-b1f4-4e27-a286-18e6a8e78a16.png)

MemeWizard replied:
> Yes.

Well, it may sound good for tribes, but it doesn't work for cults.

MemeWizard commented on localization/english/wc_laws_l_english.yml:
```diff
@@ -0,0 +1,27 @@
+﻿l_english:
+ dark_authority:0 "[dark_authority|E]"
+ dark_authority_0:0 "Disparate Cults"
+ dark_authority_1:0 "Weak Covens"
```
Maybe replace weak with something that corresponds with influence more.
```suggestion
 dark_authority_1:0 "Insignificant Circles"
```

MemeWizard commented on localization/english/wc_laws_l_english.yml:
```diff
@@ -0,0 +1,27 @@
+ dark_authority:0 "[dark_authority|E]"
+ dark_authority_0:0 "Disparate Cults"
+ dark_authority_1:0 "Weak Covens"
+ dark_authority_2:0 "Black Sabbath"
```
```suggestion
 dark_authority_2:0 "Dark Covens"
```

MemeWizard commented on localization/english/wc_laws_l_english.yml:
```diff
@@ -0,0 +1,27 @@
+ dark_authority_0:0 "Disparate Cults"
+ dark_authority_1:0 "Weak Covens"
+ dark_authority_2:0 "Black Sabbath"
+ dark_authority_3:0 "Unholy Katavasia"
```
Only russian slang uses this type of hymn as an "chaos" synonym

```suggestion
 dark_authority_3:0 "Unholy Congregations"
```

zumbak04 commented on localization/english/wc_effects_l_english.yml:
```diff
@@ -18,4 +18,14 @@
\ No newline at end of file
+ laughing_skull_trick_tooltip_first_past:0 "The Laughing Skulls slipped into the Dark Portal"
+
+ change_dark_frenzy_effect_first:0 "You gain $VALUE|+0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_first_neg:0 "You lose $VALUE|-0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_first_past:0 "You gained $VALUE|+0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_first_past_neg:0 "You lost $VALUE|-0$ [dark_frenzy|E]"
```
> I'd change these to "Your [dark_frenzy|E] changes by $VALUE|+0$ / $VALUE|-0$"

It's in the same format as money, piety, prestige. Thus you lose, you gain.

arithon replied:
Maybe, but does that make it better? If you gain something, most of the time that means +x. So saying "you gain +x" is kinda doubly matched. I'd vote for my suggested change, though your first version is not wrong per se.

zumbak04 replied:
> you gain +x

It says `You gain 100 Dark Frenzy`, there's no plus.

arithon replied:
Oh, ok. Then that's just my bad, somehow I thought that the +0$ / -0$ would somehow appear. Didn't try that ingame yet. In that case it's fine :)

zumbak04 commented on localization/english/wc_effects_l_english.yml:
```diff
@@ -18,4 +18,14 @@
+ change_dark_frenzy_effect_first_past:0 "You gained $VALUE|+0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_first_past_neg:0 "You lost $VALUE|-0$ [dark_frenzy|E]"
+ 
+ change_dark_frenzy_effect_third:0 "[CHARACTER.GetShortUIName|U] gains $VALUE|+0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_third_neg:0 "[CHARACTER.GetShortUIName|U] loses $VALUE|-0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_third_past:0 "[CHARACTER.GetShortUIName|U] gained $VALUE|+0$ [dark_frenzy|E]"
+ change_dark_frenzy_effect_third_past_neg:0 "[CHARACTER.GetShortUIName|U] lost $VALUE|-0$ [dark_frenzy|E]"
```
> I'd change these to "[CHARACTER.GetShortUIName|U]'s [dark_frenzy|E] changes by $VALUE|+0$ / $VALUE|-0$"

It's in the same format as money, piety, prestige. Thus you lose, you gain.

zumbak04 commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,18 @@
+ game_concept_grim_religion_desc:0 "A [religion|E] so dark and evil that it opposes the rest of the world. They don't seek outside allies and won't stop until they reach their ultimate goal.\n\nThese religions are:\n$EFFECT_LIST_BULLET$$deathbringer$\n$EFFECT_LIST_BULLET$$doombringer$\n$EFFECT_LIST_BULLET$$shathgral$"
+ 
+ game_concept_dark_authority:0 "Dark Authority"
+ game_concept_dark_authority_desc:0 "Replaces [crown_authority|E] for [dark_governments|E].\n\nThe $game_concept_dark_authority$ laws change automatically when they accumulate some amount of [dark_frenzy|E] and thus can't be changed manually.\n\nHigh levels of $game_concept_dark_authority$ gives [tax|E], [levy|E], [development|E] bonuses but [vassals|E] dislike it."
```
> Replaces [crown_authority|E] with [dark_governments|E]

Not with. The Dark Authority laws replace the Crown Authority for Dark Governments.

arithon replied:
Afaik "replace" goes with "with sth." or "by sth.", but not with "for sth.".

zumbak04 replied:
> Afaik "replace" goes with "with sth." or "by sth.", but not with "for sth.".

I have to google English grammar terminology but here we go.

For isn't related to the replaces. The replaces is a verb, the `crown_authority` is an object, the subject is outside the sentence.

The for marks a beginning of coordinating conjunction: `for [dark_governments|E]`.

Other examples:

> It conducts to replace contractors for long-term charter of aircraft.

> It would replace and standards for desktops interfacing with the new system.

arithon replied:
Ok, thanks for the clarification, shouldn't have started this last evening/night. Forget my suggestion to "replace"...
```suggestion
 game_concept_dark_authority_desc:0 "Replaces [crown_authority|E] for [dark_governments|E].\n\nThe $game_concept_dark_authority$ laws change automatically when enough [dark_frenzy|E] is accumulated and thus can't be changed manually.\n\nHigh levels of $game_concept_dark_authority$ grant [tax|E], [levy|E] and [development|E] bonuses, but [vassals|E] dislike it."
```

zumbak04 commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,18 @@
+ game_concept_dark_government:0 "Dark Government"
+ game_concept_dark_governments:0 "Dark Governments"
+ game_concept_dark_wo_government:0 "Dark"
+ game_concept_dark_government_desc:0 "[government_forms|E] that uses [dark_authority|E] instead of [crown_authority|E]. [levies|E] and [taxes|E] given to a [liege|E] by their [vassals|E] is determined by their [opinion|E] of said Liege. However, the Liege's [dark_authority|E] can increase the minimum amount they must provide regardless of Opinion."
```
That's from vanilla. Game concepts are capitalized.

zumbak04 commented on localization/english/wc_government_l_english.yml:
```diff
@@ -0,0 +1,12 @@
+ government_dark_frenzy_from_sieges_and_raids:0 "Gets [dark_frenzy|E] from [sieges|E] and [raids|E]"
+ government_dark_frenzy_from_conversions:0 "Gets [dark_frenzy|E] from [convert_county|E] and [promote_culture|E]"
+ 
+ government_resurrect_knights:0 "Killed enemy [knights|E] become [GetTrait('being_undead').GetName( GetNullCharacter )] and join [court|E] #F (high [dark_authority|E] increases the chance)#!"
```
Not your because it's a government description. You may be of another government.

arithon replied:
But which court do they join? It's unclear from this description, they could join the court of their (former) liege after becoming undead as well.

zumbak04 replied:
> But which court do they join? It's unclear from this description, they could join the court of their (former) liege after becoming undead as well.

Should I say something like `and join Necro Court?`

arithon replied:
I guess so. Though I can't come up with a good, short solution to this atm...

zumbak04 replied:
Done in 5931ee900df555a489e47a433b765b283337de9c.

zumbak04 commented on localization/english/wc_succession_laws_l_english.yml:
```diff
@@ -0,0 +1,5 @@
+﻿l_english:
+ dark_succession_law:1 "Same Religious Family Partition"
+ dark_succession_law_subname:0 "All Children of Same Religious Family Inherit Equally"
```
Not articles to make it shorter.

arithon replied:
alright

zumbak04 commented on localization/english/wc_succession_laws_l_english.yml:
```diff
@@ -0,0 +1,5 @@
+ dark_succession_law:1 "Same Religious Family Partition"
+ dark_succession_law_subname:0 "All Children of Same Religious Family Inherit Equally"
+ dark_succession_law_effects:1 "\n#V Under $dark_succession_law$ [succession|E], your [titles|E] will be divided equally between your children of the same [religious_family|E].#!\n\n#weak $partition_base_desc$#!"
+ partition_base_desc:2 "#weak Upon succession, all [titles|E] held by the late [ruler|E] will be divided amongst their [eligible_children|E] of the same [religious_family|E], with the [player_heir|E] always being given the [primary_title|E], [realm_capital|E] and any direct [de_jure|E] titles associated with it.\n\nYounger children will be given Titles starting with those of the same [rank|E] as  ...
```
That's from vanilla. Game concepts are capitalized.

arithon replied:
Ok, in that case:
```suggestion
 partition_base_desc:2 "#weak Upon succession, all [titles|E] held by the late [ruler|E] will be divided amongst their [eligible_children|E] of the same [religious_family|E], with the [player_heir|E] always being given the [primary_title|E], [realm_capital|E] and any direct [de_jure|E] Titles associated with it.\n\nYounger children will be given Titles starting with those of the same [rank|E] as the Primary Title; if none are available they will be given lower-ranking Titles. Your Player Heir can not be preemptively granted Titles they do not stand to inherit.\n#!"
```

arithon reviewed (commented):
Only "government_resurrect_knights" is left, the rest is fine, I think

LordSilvermane commented on localization/english/wc_battles_l_english.yml:
```diff
@@ -0,0 +1,40 @@
+
+ commander_resurrected_by_enemy_friendly_player:1 "You were slain by [TARGET_CHARACTER.GetUINameNoTooltip] and resurrected as an undead!"
+ commander_resurrected_by_enemy_friendly:0 "Commander [CHARACTER.GetUINameNoTooltip] was slain by [TARGET_CHARACTER.GetUINameNoTooltip] and resurrected as an undead!"
+ commander_resurrected_by_enemy_enemy:0 "[TARGET_CHARACTER.GetUINameNoTooltip] has slain the enemy commander, [CHARACTER.GetUINameNoTooltip], and resurrect [CHARACTER.GetHerHim] as an undead!"
```
made a minor change
```suggestion
 commander_resurrected_by_enemy_enemy:0 "[TARGET_CHARACTER.GetUINameNoTooltip] has slain the enemy commander, [CHARACTER.GetUINameNoTooltip], and resurrected [CHARACTER.GetHerHim] as an undead!"
```

LordSilvermane commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,18 @@
  game_concept_grim_religions:0 "Grim Religions"
- game_concept_grim_religion_desc:0 "A [religion|E] so dark and evil that it opposes the rest of the world. They don't seek outside allies and won't stop until they reach their ultimate goal.\n\nThese religions are:\n$EFFECT_LIST_BULLET$$deathbringer$\n$EFFECT_LIST_BULLET$$doombringer$\n$EFFECT_LIST_BULLET$$shathgral$"
\ No newline at end of file
+ game_concept_grim_religion_desc:0 "A [religion|E] so dark and evil that it opposes the rest of the world. They don't seek outside allies and won't stop until they reach their ultimate goal.\n\nThese religions are:\n$EFFECT_LIST_BULLET$$deathbringer$\n$EFFECT_LIST_BULLET$$doombringer$\n$EFFECT_LIST_BULLET$$shathgral$"
```
Made a minor change
```suggestion
 game_concept_grim_religion_desc:0 "A [religion|E] so dark and evil that it opposes the rest of the world. They refuse to seek outside allies and will never stop until they reach their ultimate goal.\n\nThese religions are:\n$EFFECT_LIST_BULLET$$deathbringer$\n$EFFECT_LIST_BULLET$$doombringer$\n$EFFECT_LIST_BULLET$$shathgral$"
```

LordSilvermane commented on localization/english/wc_government_l_english.yml:
```diff
@@ -0,0 +1,12 @@
+ necro_government:0 "Necro"
+ necro_government_adjective:0 "Necro"
+ necro_government_realm:0 "Necro Realm"
+ necro_government_desc:0 "\n$game_concept_necro_government_desc$\n\n#F A realm ruled by the dead. Violent by nature and incapable of reproduction, they have to kill the living so they will join them in the afterlife.#!"
```
Changed the latter part.
```suggestion
 necro_government_desc:0 "\n$game_concept_necro_government_desc$\n\n#F A realm ruled by the dead. Violent by nature and incapable of reproduction, they have to kill and resurrect the living in order to replenish their numbers.#!"
```

LordSilvermane reviewed (commented):
Only a few suggestions, everything else looked good.

zumbak04 commented on localization/english/wc_government_l_english.yml:
```diff
@@ -0,0 +1,12 @@
+ government_dark_frenzy_from_sieges_and_raids:0 "Gets [dark_frenzy|E] from [sieges|E] and [raids|E]"
+ government_dark_frenzy_from_conversions:0 "Gets [dark_frenzy|E] from [convert_county|E] and [promote_culture|E]"
+ 
+ government_resurrect_knights:0 "Killed enemy [knights|E] become [GetTrait('being_undead').GetName( GetNullCharacter )] and join [court|E] #F (high [dark_authority|E] increases the chance)#!"
```
> Only this is left, rest is good imho.

Done in 5931ee900df555a489e47a433b765b283337de9c.

zumbak04 replied:
Done in 5931ee900df555a489e47a433b765b283337de9c.

LordSilvermane commented on localization/english/wc_laws_l_english.yml:
```diff
@@ -0,0 +1,27 @@
+﻿l_english:
+ dark_authority:0 "[dark_authority|E]"
+ dark_authority_0:0 "Scattered Cults"
+ dark_authority_1:0 "Insignificant Circles"
```
One very minor quibble, shouldn't these two be switched? Scattered cults sound more influential than insignificant circles to me. I mean, a circle is defined as a group of people, whereas a cult is defined using words like organization, religion, and system of beliefs and their adherents, which sounds more powerful than the former. 

Just my opinion though, and not very important. Thought I'd point it out anyways. Rest looks good.

arithon replied:
You got a point there that I'd like to support :)

zumbak04 replied:
> One very minor quibble, shouldn't these two be switched? Scattered cults sound more influential than insignificant circles to me. I mean, a circle is defined as a group of people, whereas a cult is defined using words like organization, religion, and system of beliefs and their adherents, which sounds more powerful than the former.
> 
> Just my opinion though, and not very important. It works as is, just a question of terminology. Thought I'd point it out anyways. Rest looks good.

Done in 4cddafbe66b728a5dc7a360a06af9ba2cf172115.

LordSilvermane reviewed (approved):
All looks good. 

I made one more suggestion, but it isn't necessary or important. Figured I'd point it out when I noticed, though.

IronyOwl reviewed (approved):
Looks good to me.

Slight oddity: I got an event where a councilor asked me to host a hunt or feast soon. If we're keeping that vanilla event, presumably we'd want to stop it from firing (or replace it with sacrifices) for Grim religion rulers.

zumbak04 commented:
> Slight oddity: I got an event where a councilor asked me to host a hunt or feast soon. If we're keeping that vanilla event, presumably we'd want to stop it from firing (or replace it with sacrifices) for Grim religion rulers.

What event ID?

zumbak04 commented:
> Slight oddity: I got an event where a councilor asked me to host a hunt or feast soon. If we're keeping that vanilla event, presumably we'd want to stop it from firing (or replace it with sacrifices) for Grim religion rulers.

Disabled in c6158edb940f8e42a473bf5bc024ce90d255e62f.

arithon reviewed (approved):
loc LGTM

TheGroxEmpire reviewed (commented):
All features seems to be working as intended:
- Dark Frenzy are changed appropriate to Frenzy level.
- Frenzy are earned from battles, sieges, and raids.
- Slain knight are sometimes revived and appeared in court. All of their living relatives / relationship seems to be wiped off too.
- Only ruler that is the same family religion can inherit Dark Government titles.
- Undead is a crime for most religion: allows revocation and imprisonment without tyranny
- New Lich King / Queen are appointed with their appropriate titles.

There was a single bug. If you converted a Dark Government ruler faith into something other than Deathbringer faith, It'll destroy Dark Government into regular government and disabled any future Dark Government even if it would be inherited by Deathbringer ruler. Here I converted Scourge ruler into Holy Light. After I killed the converted ruler it was inherited by Deathbringer ruler. Yet the government stays Feudal after inheritance:
<details>

![image](https://user-images.githubusercontent.com/47816228/127988238-04629c0c-1424-4f32-bff5-1044b485d58e.png)
<details>

I also think that the decay is currently too low. I think it would be more interesting if decay scales up with your realm size.

zumbak04 commented:
> There was a single bug. If you converted a Dark Government ruler faith into something other than Deathbringer faith, It'll destroy Dark Government into regular government and disabled any future Dark Government even if it would be inherited by Deathbringer ruler. Here I converted Scourge ruler into Holy Light. After I killed the converted ruler it was inherited by Deathbringer ruler. Yet the government stays Feudal after inheritance:
Fixed in ef9ad4183412d2174ae6f838a70c72989d490ec3.

> I also think that the decay is currently too low. I think it would be more interesting if decay scales up with your realm size.
Added in 32ffa0b912631e0886d296c333adaee28e4b9275.

TheGroxEmpire reviewed (commented):
> Fixed in ef9ad41.

Can confirm this does fix the government inheritance issue. This fix also has side effect that Lich King cannot adopt other faith even through console command, though I don't really mind with this.

> Added in 32ffa0b.

The decay scaling with realm size has been implemented. However, it looks like the realm size is based on liege realm size rather than ruler realm size.
<details>
![image](https://user-images.githubusercontent.com/47816228/128133500-982efd35-417c-44c8-bc87-752a13af0644.png)

![image](https://user-images.githubusercontent.com/47816228/128133537-9431488a-bbfe-4e15-8447-92e987e65415.png)

</details>

zumbak04 commented:
> The decay scaling with realm size has been implemented. However, it looks like the realm size is based on liege realm size rather than ruler realm size.

Fixed in 2eb1417e0e84d3773a1d4b4e04165d2ab6a64e41.

TheGroxEmpire reviewed (approved):
My issues have been addressed. Everything looks good to me!
